components/asana/asana.app.js

```diff
@@ -0,0 +1,251 @@
+const axios = require("axios");
+const crypto = require("crypto");
+
+module.exports = {
+  type: "app",
+  app: "asana",
+  propDefinitions: {
+    workspaceId: {
+      type: "string",
+      label: "Workspace",
+      optional: false,
+      async options(opts) {
+        const workspaces = await this.getWorkspaces();
+        return workspaces.map((workspace) => {
+          return {
+            label: workspace.name,
+            value: workspace.gid,
+          };
+        });
+      },
+    },
+    projectId: {
+      type: "string",
+      label: "Project",
+      optional: false,
+      async options(opts) {
+        const projects = await this.getProjects(opts.workspaceId);
+        return projects.map((project) => {
+          return {
+            label: project.name,
+            value: project.gid,
+          };
+        });
+      },
+    },
+    taskIds: {
+      type: "string[]",
+      label: "Tasks",
+      async options(opts) {
+        const tasks = await this.getTasks(opts.projectId);
+        return tasks.map((task) => {
+          return { label: task.name, value: task.gid };
+        });
+      },
+    },
+    organizationId: {
+      type: "string",
+      label: "Organization",
+      optional: false,
+      async options(opts) {
+        const organizations = await this.getOrganizations();
+        return organizations.map((organization) => {
+          return {
+            label: organization.name,
+            value: organization.gid,
+          };
+        });
+      },
+    },
+  },
+  methods: {
+    async _getHeaders() {
+      return {
+        Accept: "application/json",
+        Authorization: `Bearer ${this.$auth.oauth_access_token}`,
+      };
+    },
+    async _getAuthorizationHeader({ data, method, url, headers }) {
+      const token = {
+        key: this.$auth.oauth_access_token,
+        secret: this.$auth.oauth_refresh_token,
+      };
+      return await axios({
+        method: "POST",
+        url: `https://app.asana.com/api/1.0/webhooks`,
```

Comment:
Since you reference this base URL in a few places, you might create a method like this:

```javascript
_getBaseURL() {
  return "https://app.asana.com/api/1.0/"
}
```

and reference that everywhere you use the URL.